docs/application-development-guide.md

```diff
@@ -1771,7 +2108,123 @@ workflowRun, err := c.ExecuteWorkflow(context.Background(), workflowOptions, "Yo
+  ```
+
+- Asynchronous start initiates a Workflow Execution and immediately returns to the caller. This is the most common way to start Workflows in production code.
+  The `WorkflowClient`<https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/client/WorkflowClient.java)> provides some static methods, such as `start`, `execute`, `signalWithStart` etc., that help with starting your Workflows asynchronously.
+
+  The following examples show how to start Workflow Executions asynchronously, with either typed or untyped `WorkflowStub`.
+
+  - **Typed WorkflowStub Example**
+
+    ```java
+    // create typed Workflow stub
+    FileProcessingWorkflow workflow = client.newWorkflowStub(FileProcessingWorkflow.class,
+          WorkflowOptions.newBuilder()
+                  .setTaskQueue(taskQueue)
+                  .setWorkflowId(workflowId)
+                  .build());
+    // use WorkflowClient.execute (if your Workflow takes in arguments) or WorkflowClient.start (for zero arguments)
+    WorkflowClient.start(workflow::greetCustomer);
+    ```
+
+  - **Untyped WorkflowStub Example**
+
+    ```java
+    WorkflowStub untyped = client.newUntypedWorkflowStub("FileProcessingWorkflow",
+          WorkflowOptions.newBuilder()
+                  .setWorkflowId(workflowId)
+                  .setTaskQueue(taskQueue)
+                  .build());
+
+    // blocks until Workflow Execution has been started (not until it completes)
+    untyped.start(argument);
+    ```
+
+You can call a Dynamic Workflow implementation using an untyped `WorkflowStub`.
+The following example shows how to call the Dynamic Workflow implementation in the Client code.
+
+```java
+    WorkflowClient client = WorkflowClient.newInstance(service);
+    /**
+      * Note that for this part of the client code, the dynamic Workflow implementation must
+      * be known to the Worker at runtime in order to dispatch Workflow tasks, and may be defined
+      * in the Worker definition as:*/
+    // worker.registerWorkflowImplementationTypes(DynamicGreetingWorkflowImpl.class);
+
+    /* Create the Workflow stub to call the dynamic Workflow.
+    * Note that the Workflow type is not explicitly registered with the Worker.*/
+    WorkflowOptions workflowOptions =
+        WorkflowOptions.newBuilder().setTaskQueue(TASK_QUEUE).setWorkflowId(WORKFLOW_ID).build();
+    WorkflowStub workflow = client.newUntypedWorkflowStub("DynamicWF", workflowOptions);
+```
+
+`DynamicWorkflow` can be used to invoke different Workflow Types.
+To check what type is running when your Dynamic Workflow `execute` method runs, use `getWorkflowType()` in the implementation code.
+
+```java
+String type = Workflow.getInfo().getWorkflowType();
+```
+
+See [Workflow Execution Result](#get-workflow-results) for details on how to get the results of the Workflow Execution.
+
+#### Using `ExternalWorkflowStub`
```

Comment:
Can't have these headers in the source content.

Reply:
updated across.